src/scripts/main.js

```diff
@@ -1,3 +1,33 @@
 'use strict';
 
-// write code here
+const array = [...document.querySelectorAll('li')];
+
+const toNum = (el) => {
+  return parseFloat(el.replace(/[$,]/g, ''));
+};
+
+const sortList = (list) => {
+  list.sort((a1, a2) => toNum(a2.dataset.salary) - toNum(a1.dataset.salary));
```

Comment:
Ensure that each list item has a 'data-salary' attribute. If any item is missing this attribute, `toNum` will return NaN, which could lead to unexpected sorting behavior.